### PR TITLE
Updates and upgrades

### PR DESCRIPTION
**Pull request by ksatirli**
Hello reviewers 👋🏼 

This PR carries out some basic maintenance, adds an example and updates the code to be in line with other Terraform modules.

### CODE REVIEW

File: examples/basic/.terraform.tfstate.lock.info [hunk 1]
```diff
@@ -0,0 +1 @@
+{"ID":"aa5059a6-2eb2-7bc4-6a53-c12f2aca7d60","Operation":"OperationTypeApply","Info":"","Who":"ksatirli@ksatirli-F5KDW0LRK7GF","Version":"1.2.8","Created":"2022-08-26T14:27:19.344016Z","Path":"terraform.tfstate"}
```
Comment (straubt1) [inline examples/basic/.terraform.tfstate.lock.info]: Is this file needed?

File: main.tf [hunk 2]
```diff
@@ -15,33 +17,37 @@ resource "aws_route53_record" "autodiscover" {
...
+resource "aws_ses_domain_dkim" "main" {
+  domain = aws_ses_domain_identity.main.domain
 }
 
+# see https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record
 resource "aws_route53_record" "dkim" {
   count   = 3
   zone_id = var.zone_id
-  name    = "${element(aws_ses_domain_dkim.dkim.dkim_tokens, count.index)}._domainkey.${local.zone_name}"
+  name    = "${element(aws_ses_domain_dkim.main.dkim_tokens, count.index)}._domainkey.${local.zone_name}"
```
Comment (straubt1) [inline main.tf:44]: You could index directly instead of using `element` which is a TF 0.11 type syntax.

Suggested:
```
records = ["${aws_ses_domain_dkim.main.dkim_tokens[count.index]}.dkim.amazonses.com."]
```

**Review by straubt1 — APPROVED** [review]
A few minor suggestions but nothing that breaks this module.
LGTM!

File: main.tf (hunk 2 above)
Comment (ksatirli) [reply main.tf:44]: Nope - removed and ignored!
